src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -157,30 +159,67 @@ private BoundExpression MakeIndexerAccess(
                 Debug.Assert(getMethod is not null);
 
                 ArrayBuilder<LocalSymbol>? temps = null;
+                bool needSpecialExtensionPropertyReceiverReadOrder = false;
+                ArrayBuilder<BoundExpression>? storesOpt = null;
+
+                if (IsPropertyWithByValPossiblyStructReceiverWhichHasLocationAndCanChangeValueBetweenReads(rewrittenReceiver, indexer))
+                {
```

Comment:
Consider leaving follow-up comment to cover this logic when implementing extension indexers